docs/coding-guidelines.md

```diff
@@ -0,0 +1,403 @@
+
+1. We use four spaces of indentation (no tabs).
+
+1. We use `_camelCase` for internal and private fields and use `readonly` where possible. Prefix internal and private instance fields with `_`. Static fields are all CamelCase regardless of visibility. When used on static fields, `readonly` should come after `static` (e.g. `static readonly` not `readonly static`).  Public fields should be used sparingly and should use PascalCasing with no prefix when used.
+
+1. We avoid `this.` unless absolutely necessary.
+
+1. We always specify the visibility, even if it's the default (e.g.
+   `private string _foo` not `string _foo`). Visibility should be the first modifier (e.g.
+   `public abstract` not `abstract public`).
+
+1. Namespace imports should be specified at the top of the file, *outside* of
+   `namespace` declarations, and should be sorted alphabetically, with the exception of `System.*` namespaces, which are to be placed on top of all others.
+
+1. Avoid more than one empty line at any time. For example, do not have two
+   blank lines between members of a type.
+1. Avoid spurious free spaces.
+   For example avoid `if (someVar == 0)...`, where the dots mark the spurious free spaces.
+   Consider enabling "View White Space (Ctrl+R, Ctrl+W)" or "Edit -> Advanced -> View White Space" if using Visual Studio to aid detection.
+
+1. If a file happens to differ in style from these guidelines (e.g. private members are named `m_member`
+   rather than `_member`), the existing style in that file takes precedence. Changes/refactorings are possible, but depending on the complexity, change frequency of the file, might need to be considered on their own merits in a separate pull request.
+
+1. We only use `var` when it's obvious what the variable type is.
+For example the following are correct:
+
+    ```cs
+    var fruit = "Apple";
+    var fruits = new List<Fruit>();
+    string fruit = null; // can't use "var" because the type isn't known (though you could do (string)null, don't!)
+    const string expectedName = "name"; // can't use "var" with const
+    FruitFlavor flavor = fruit.GetFlavor();
+    ```
+
+    The following are incorrect:
+
+    ```cs
+    var flavor = fruit.GetFlavor();
+    string fruit = "Apple";
+    List<Fruit> fruits = new List<Fruit>();
+    ```
+
+1. We use language keywords instead of BCL types (e.g. `int, string, float` instead of `Int32, String, Single`, etc) for both type references as well as method calls (e.g. `int.Parse` instead of `Int32.Parse`).
+    The following are correct:
+
+    ```cs
+    public string TrimString(string s) {
+        return string.IsNullOrEmpty(s)
+            ? null
+            : s.Trim();
+    }
```

Comment:
I suggest removing the body of the method, as it doesn't add anything to the example (it doesn't contain any type names), and therefore is identical to the incorrect example, making it harder to see the differences.

Reply:
It does contain type names: 

string.IsNullOrEmpty(s) vs
String.IsNullOrEmpty(s)